tests/test_coordUtils.py

```diff
@@ -0,0 +1,325 @@
+            ):
+                for shift3 in (-550, 0, 375):
+                    predictedShiftedPlanePos = [planePos[i] + shift3*ratios[i] for i in range(2)]
+                    shiftedPlanePos = coordUtils.computeShiftedPlanePos(planePos, fieldAngle, shift3)
+                    self.assertPairsAlmostEqual(shiftedPlanePos, predictedShiftedPlanePos)
+
+    def testConvertVectorFromPupilToBase(self):
+        """Test convertVectorFromPupilToBase and convertVectorFromBaseToPupil
+        """
+        cos30 = math.cos(30 * math.pi / 180)
+        sin30 = math.sin(30 * math.pi / 180)
+        for magMultiplier in (0.001, 1, 1000):
+            for azAltDeg, vectorPupil, predictedVectorBase in (
+                # at az=0, alt=0: base = pupil
+                ((0, 0), (1, 0, 0), (1, 0, 0)),
+                ((0, 0), (0, 1, 0), (0, 1, 0)),
+                ((0, 0), (0, 0, -1), (0, 0, -1)),
+                ((0, 0), (1, -1, 1), (1, -1, 1)),
+                # at az=90, alt=0: base x = pupil -y, base y = pupil x, base z = pupil z
+                ((90, 0), (1, 0, 0), (0, 1, 0)),
+                ((90, 0), (0, 1, 0), (-1, 0, 0)),
+                ((90, 0), (0, 0, -1), (0, 0, -1)),
+                ((90, 0), (1, -1, 1), (1, 1, 1)),
+                # at az=0, alt=90: base x = - pupil z, base y = pupil y, base z = pupil x
+                ((0, 90), (1, 0, 0), (0, 0, 1)),
+                ((0, 90), (0, 1, 0), (0, 1, 0)),
+                ((0, 90), (0, 0, -1), (1, 0, 0)),
+                ((0, 90), (1, -1, 1), (-1, -1, 1)),
+                # at az=90, alt=90: base x = -pupil y, base y = - pupil z, base z = pupil x
+                ((90, 90), (1, 0, 0), (0, 0, 1)),
+                ((90, 90), (0, 1, 0), (-1, 0, 0)),
+                ((90, 90), (0, 0, -1), (0, 1, 0)),
+                ((90, 90), (1, -1, 1), (1, -1, 1)),
+                # at az=0, alt=45:
+                # base x = cos(30) * pupil x - sin(30) * pupil z
+                # base y = pupil y
+                # base z = sin(30) * pupil x + cos(30) * pupil z
+                ((0, 30), (1, 0, 0), (cos30, 0, sin30)),
+                ((0, 30), (0, 1, 0), (0, 1, 0)),
+                ((0, 30), (0, 0, -1), (sin30, 0, -cos30)),
+                ((0, 30), (1, -1, 1), (cos30 - sin30, -1, sin30 + cos30)),
+                # at az=30, alt=0:
+                # base x = cos(30) * pupil x - sin(30) * pupil y
+                # base y = sin(30) * pupil x + cos(30) * pupil y
+                # base z = pupil z
+                ((30, 0), (1, 0, 0), (cos30, sin30, 0)),
+                ((30, 0), (0, 1, 0), (-sin30, cos30, 0)),
+                ((30, 0), (0, 0, -1), (0, 0, -1)),
+                ((30, 0), (1, -1, 1), (cos30 + sin30, sin30 - cos30, 1)),
+            ):
+                vectorPupil = np.array(vectorPupil) * magMultiplier
+                predictedVectorBase = np.array(predictedVectorBase) * magMultiplier
+                pupilAzAlt = SpherePoint(*azAltDeg, degrees)
```

Comment:
I'm highly dubious that this is a problem, but it would be remiss of me not to mention that my linter says this is an "E901 syntax error: Only named arguments may follow *expression". I imagine this is a py2/3 things, and my setup is stuck in the past or something, but mentioning it just in case.

Reply:
Is your linter using python2? (look at the shebang line).